brion/plugin/compartmentReportHDF5.cpp

```diff
@@ -125,71 +126,37 @@ CompartmentReportHDF5::CompartmentReportHDF5(
 
         if (accessMode & MODE_WRITE)
         {
-            if ((accessMode & MODE_OVERWRITE) != MODE_OVERWRITE &&
-                fs::exists(_path))
+            const bool exists = fs::exists(_path);
+            if (exists && (accessMode & MODE_OVERWRITE) != MODE_OVERWRITE)
             {
                 LBTHROW(std::runtime_error("Cannot overwrite existing file " +
                                            _path.string()));
             }
+            const auto writeFlag =
+                exists ? HighFive::File::Truncate : HighFive::File::Excl;
+            _file.reset(new HighFive::File(_path.string(),
+                                           HighFive::File::Create | writeFlag));
 
-            _file = H5::H5File(_path.string(), H5F_ACC_TRUNC);
             _reportName = fs::basename(_path);
             return;
         }
 
-        H5::H5File file;
-
         // assume one file per cell
         if (fs::is_directory(_path))
         {
             fs::directory_iterator it(_path);
             if (it == fs::directory_iterator())
                 LBTHROW(std::runtime_error("No files in " + _path.string()));
-            file = H5::H5File(it->path().string(), H5F_ACC_RDONLY);
-            if (!file.getId())
-                LBTHROW(std::runtime_error("File not found: " +
-                                           it->path().string()));
+            HighFive::File file(it->path().string(), HighFive::File::ReadOnly);
```

Comment:
you get similar throws from highfive if file was not found?

Reply:
I'll review the errors being printed. I should also silence the stack traces from HDF5

Reply:
The errors are probably OK like this.